M7md023x power detail design

m7md023x opened the pull request:
The design is completely changed.

m7md023x committed:
Add files via upload

m7md023x committed:
Add files via upload

m7md023x committed:
Add files via upload

m7md023x committed:
detaildesign_power.md

m7md023x committed:
Update_detaildesign power.md

m7md023x committed:
Add files via upload

m7md023x committed:
lastUpdate_detaildesign_power.md

m7md023x committed:
Detail_design_subsystem

ambacon43 commented:
Hey Mohammad, good start on this. A few formatting notes: 
1) missing a space in first paragraph "for delivering"
2) some heading stuff is not emboldened right (I think you need a space before the ###)
3) your final BOM item didn't hyperlink
4) just polish the analysis section, as far as capitalization & punctuation goes

Overall quality material and nice schematics. Good work!

MaddoxCagle commented:
Hey Mohammad, here’s my notes regarding your detailed design.
1. I agree with everything Mandy put, so I am not going to rewrite it all in mine.
2. What is the voltage boost rated for? Is it a percentage of the supplied voltage, or is it just a set voltage regardless of the input voltage?
3. What charges the USB power bank?
4. How does 70 hours meet the “72-hour minimum operational target per charge cycle”.
5. I did read about the improvement investigation after typing the above comment, so it is good that you are doing that.
6. I really enjoyed your detailed design. I think you may could go into some more detail of each subsystem to answer some of these questions above, but a great start for sure!

MaddoxCagle reviewed (approved):
Looks great!

m7md023x committed:
Updated 04/24 detaildesign power.md

m7md023x committed:
Update detaildesign power.md

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,128 @@
+This design uses:
+- 3-cell AA Eneloop Pro battery packs with MT3608 boost converters for ESP32 units.
+- A Geekworm X728 UPS HAT with two 20,000mAh USB-C power banks and two 18650 batteries to power the Raspberry Pi.
+- The system enables continuous off-grid operation with safe switchover and recharge methods.
```
Should this be on the list? It is not a device.

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,128 @@
+|------|------------------------------------|------------------|-----------------|----------------------|-----|-------------|-----------|----------------------------------------------------------------------|
+| U1   | MT3608 Boost Converter             | MT3608           | AITRIP          | Amazon               | 3   | $1.20       | $3.60     | [Link](https://www.amazon.com/dp/B0C858YYQ1)                         |
+| B1   | Eneloop Pro AA Battery (10-Pack)   | BK-3HCCA10FA     | Panasonic       | Amazon               | 10  | $45.99      | $45.99    | [Link](https://www.amazon.com/dp/B0D2JBNH4H)                         |
+| H1   | 3-AA Battery Holder                | BH3AAW           | Elenco          | Digikey              | 3   | $1.50       | $4.50     | [Link](https://www.digikey.com)                                     |
+| D1   | LED (3mm Red)                      | LTL-307EE        | Lite-On         | Digikey              | 3   | $0.50       | $1.50     | [Link](https://www.digikey.com)                                     |
+| R1   | Resistor 330Ω                      | CF14JT330R       | Stackpole       | Mouser               | 3   | $0.05       | $0.15     | [Link](https://www.mouser.com)                                      |
+| P1   | JST Connector Pair                 | B2B-PH-K-S       | JST             | Digikey              | 12  | $0.30       | $3.60     | [Link](https://www.digikey.com)                                     |
```
Link to the product not just the distributor's website.

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,128 @@
+
+###7. Printed Circuit Board Layout
+![Schematic](Printed_Circuit_Board_Layot.png)
+
```
You need to add a connection block diagram so that I can see what connects to what.

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,128 @@
+
+###5. 3D Model
+
+![3dmodel](3D_Model.png)
+
+###6. Buildable Schematic
+![Schematic](Buildable_circuit.png)
+
+###7. Printed Circuit Board Layout
+![Schematic](Printed_Circuit_Board_Layot.png)
```
Would something off-the-shelf work? It would probably be cheaper. Like this:

https://www.amazon.com/Diymore-Lithium-Battery-Charging-Arduino/dp/B07SZKNST4/ref=sr_1_14?dib=eyJ2IjoiMSJ9.I8X4VHN5473Swh9-S8ogKohsbM41jYkg6tFeuDOqs5Zlt_vv3nz4bJYew4wyphDnK0HY75f3v0kDfzeyJLeQOLnOcsmImhJ3dYv6zVD3moH_YveWZMLSic3c5GaDW3UytzDckxdyM_lbTRt8SzOmNTcYF__24ACMUAKjbD8wTb79u6rdA2AysoxyDoDbijiQD4uhDTEa0vzYNGhEmd9G40LGGmaXaPbLC0arx7wqWm8.pXplX1fkXjA2tOanJYqGMyg8TnwYe8KjNZmrZN0Gq0E&dib_tag=se&keywords=esp32%2Bbattery&qid=1745605187&sr=8-14&th=1

micahr234 replied:
What did you do to address this?

m7md023x replied:
I used the suggested off-the-shelf to power the raspberry pi, but i kept the same olde method to power the the esp32.   I think i misunderstood what you were asking. But, I do believe the new detailed design is cheaper which achieve the desired goal.

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,128 @@
+---
+
+## 9. Analysis
+
```
This section should focus on engineering analysis.

How long will the batters last: for the ESP32 and sensors? for the PI?

Can the batteries supply enough current for maximum draw?

m7md023x committed:
Add files via upload

m7md023x committed:
Update 04/26 detaildesign power.md

m7md023x committed:
secondUpdate 04/26 detaildesign power.md

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,252 @@
+
+### Raspberry Pi 4 Power Consumption
+
+| Condition | Current Draw | Power Consumption |
+|:---|:---|:---|
+| Light/Idle Load | 700mA @ 5V | 3.5W (5V × 0.7A) |
+| Moderate Load | 1.5A @ 5V | 7.5W (5V × 1.5A) |
```
Where did you get these numbers for the rpi? This would be a good place to add references.

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,252 @@
+
+| Load Condition | Runtime (hours) |
+|:---|:---|
+| Light/Idle (700mA) | 6.29 hours (22Wh / 3.5W) |
+| Moderate Load (1.5A) | 2.93 hours (22Wh / 7.5W) |
```
Does this meet your specifications?

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,252 @@
+
+### ESP32 and Sensor Load
+
+| Component | Typical Current Draw | Peak Current Draw |
+|:---|:---|:---|
+| ESP32 (Wi-Fi Idle) | ~80mA | ~500mA (transmitting) |
+| BH1750 Light Sensor | 120µA | - |
+| BME280 Temp/Humidity/Pressure Sensor | 3.6µA | 714µA peak |
+| Gravity O₂ Sensor | ~100µA | - |
+| SCD40 CO₂ Sensor | 15mA avg | 205mA peak |
```
References?

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,252 @@
+
+| Load Condition | Runtime (hours) |
+|:---|:---|
+| Typical Load (150mA) | 10.4 hours (7.8Wh / (5V × 0.15A)) |
```
Does this meet specifications?

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,252 @@
+| Raspberry Pi 4 | Diymore 18650 Shield (2×3500mAh 18650 cells) | 6.29h light load, 2.93h moderate load | ✅ Enough for light/moderate loads |
+| ESP32 + Sensors | 3x AA Eneloop Pro + MT3608 Boost | 10.4h typical | ✅ Safely supports normal operation |
+
+Thus, the proposed power design is fully appropriate for the intended lightweight greenhouse environmental monitoring system.
```
I am confused. I see your detailed design specification says:

Each sensor module shall operate for a typical duration of 10–12 hours per full charge cycle (it will depend on other subsystems to reach more than 72 hours).

But your conceptual design says:

Batteries shall be able to operate for minimum of 72 hours before recharging.

Which is it?

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,252 @@
+**Total BOM Cost: $140.31**
+## 10. Analysis
+
+# Power System Analysis
```
MUCH better analysis!

m7md023x committed:
04/28 detaildesign power.md

This is a fully updated version of the detail design using a different approach to build the system.

m7md023x committed:
Add files via upload

m7md023x committed:
04/28 detaildesign power.md

fully updated design

m7md023x committed:
04/28 detaildesign power.md

fully updated design

m7md023x commented:
please ignore the old graphs and block diagram, I've deleted it but it still there. there is a new block diagram.

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,248 @@
+
+## 2. Specifications and Constraints
+
+### Specifications
+
+1. The subsystem **shall** provide regulated output voltages of 5V depending on module requirements.
+2. Each ESP32 sensor module **shall** be powered by a Diymore 18650 V9 battery shield with four Samsung 30Q 3000mAh 18650 cells.
+3.Each ESP32 sensor module **shall** operate for a minimum of 72 hours per charge cycle.
+4. The Raspberry Pi 4 **shall** be powered by an ECO-WORTHY 12V 50Ah LiFePO₄ battery via a DROK buck converter that outputs regulated 5V via USB.
+5. The power output to the Raspberry Pi 4 **shall** be delivered through a USB-A to USB-C cable.
+6. The battery system **shall** support over 72 hours of Raspberry Pi 4 runtime per charge.
+
+### Constraints
+
+1. The subsystem **shall** not exceed 50V DC under any condition, in compliance with OSHA Standard 29 CFR 1910.303.
+2. All components **shall** operate safely within a temperature range of 0°C to 60°C.
+3. All electrical materials and connectors **shall** meet ASTM B117 corrosion resistance requirements.
```
You are confusing specifications (given by your team in conceptual design) and your solution. I do not want the solution here, instead please give the specifications and constraints that govern your solution.

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,248 @@
+|:---|:---|:---|
+| Typical Load | 600mA @ 5V | 3.0W |
+| Peak Load | 1.2A @ 5V | 6.0W |
+[https://www.raspberrypi.com/documentation/computers/raspberry-pi.html]
```
do not put a url, put a number that i can find in your references

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,248 @@
+Runtime = 480Wh ÷ 6.0W = **80 hours**
+
+**Conclusion**:  
+The Raspberry Pi 4 can operate for approximately **160 hours** at typical usage and **80 hours** under full load. The ECO-WORTHY 50Ah 12V battery with a DROK buck converter safely supports both conditions.
```
Have you discussed using the ECO-WORTHY 50Ah 12V battery with your customer? Are they ok with this? They would have to recharge. Do they have a way of easily doing that?

m7md023x replied:
After our presentation with the customer and explaining the solution we had, she decided to let us use the wall power and avoid long wiring so the solution i presented on the updated design meet those specifications.

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,248 @@
+**Runtime Estimation**:
+
+- Power consumption (typical): 5V × 0.11A = **0.55W**
+- Usable battery energy: 5V × 9.6Ah = **48Wh**
```
Should be 3.6v x 9.6Ah. The amp hour rating of the battery is based on the battery voltage not the output voltage of the regulator.

m7md023x committed:
04/30 detaildesign power.md

m7md023x committed:
04/30 detaildesign power.md

m7md023x committed:
05/02Update detaildesign power.md

The new design is approved by the customer

m7md023x committed:
Add files via upload

m7md023x committed:
05/02Update detaildesign power.md

Customer approved the new design

m7md023x commented:
The customer changed her specifications and allowed us to use the wall power.

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,273 @@
+- Usable battery energy: 3.6V × 9.6Ah = **34.56Wh**
+- Estimated runtime: 34.56Wh ÷ 0.55W ≈ **62.8 hours**
+
+With deep sleep enabled, the ESP32 draws less power and exceeds the 72-hour minimum runtime.
```
Are you hoping that this is true or can you prove it?

micahr234 commented on detaildesign power.md:
```diff
@@ -0,0 +1,273 @@
+
+**Conclusion**:  
+The Vilros Plug 'n' Pi kit is a reliable and streamlined solution for powering the Raspberry Pi 4. It ensures sufficient current delivery, reduces design complexity, and supports continuous operation — making it the preferred choice for the final design and met the customer specifications.
+
```
Is there a plug inside the green house? If not, does your customer understand the range limitations of wifi?

m7md023x replied:
the plug is inside the greenhouse according to the customer, and she agreed to use it.

m7md023x committed:
05/03 detaildesign power.md

m7md023x committed:
05/03 detaildesign power.md

m7md023x committed:
05/03 Update detaildesign power.md
